### PR TITLE
Fix leaking parcel file descriptors

### PR DESCRIPTION
**Pull request by Unpublished**
Caused `A resource failed to call close` warnings to be logged.

Found via [StrictMode](https://developer.android.com/reference/android/os/StrictMode)

### CODE REVIEW

**Comment by stefan-niedermann** [comment]
Thank you for the contribution! Really looking forward to it [since we also experiencing this issue](https://github.com/stefan-niedermann/nextcloud-deck/issues/539) :partying_face:

**Comment by David-Development** [comment]
Great work, thank you! 🙏

**Review by David-Development — APPROVED** [review]

**Review by Unpublished — COMMENTED** [review]
Hold on, I probably found another one

File: src/main/java/com/nextcloud/android/sso/api/AidlNetworkRequest.java [hunk 1]
```diff
@@ -164,16 +163,21 @@ private void waitForApi() throws NextcloudApiNotRespondingException {
...
+                if (response.getException().getMessage() != null) {
+                    throw parseNextcloudCustomException(response.getException());
+                }
+                throw response.getException();
             }
-            throw response.getException();
+            return new Response(os, response.headers);
+        } catch (Exception e) {
+            os.close();
+            throw e;
```
Comment (Unpublished) [inline src/main/java/com/nextcloud/android/sso/api/AidlNetworkRequest.java:179]: Don't like this...I'm open to suggestions

Comment (tobiasKaminsky) [reply src/main/java/com/nextcloud/android/sso/api/AidlNetworkRequest.java:179]: Remove catch and put os.close() in finally?
Then Exception will be thrown, but also os.close() is called?

(not tested, but should work)

Comment (Unpublished) [reply src/main/java/com/nextcloud/android/sso/api/AidlNetworkRequest.java:179]: That was my initial solution. Then I noticed `os` is returned and is intended to stay open, it only needs to be closed on an exception when it isn't returned anymore.

Comment (tobiasKaminsky) [reply src/main/java/com/nextcloud/android/sso/api/AidlNetworkRequest.java:179]: Ah. Then I think the solution is good.
Can you add an info that os normally should stay open, but only be closed when handling an Exception?

Comment (Unpublished) [reply src/main/java/com/nextcloud/android/sso/api/AidlNetworkRequest.java]: Something like [this](https://github.com/nextcloud/Android-SingleSignOn/compare/e1b80f18df5b453ea2367937fe004ba0ebfbae87..4b1a3cb821f61b3ac604b54a102efcf5ef04a6ca)?

**Comment by tobiasKaminsky** [comment]
> Hold on, I probably found another one

Is this still valid?

**Review by Unpublished — COMMENTED** [review]
Nope, this is ready to be merged :)

**Pull request merged**